tests/vanilla/store.test.tsx

```diff
@@ -225,3 +225,45 @@ it('should update async atom with deps after await (#1905)', async () => {
   expect(await lastValue).toBe(3)
   unsub()
 })
+
+it('should not fire subscription when async atom promise is the same', async () => {
```

Comment:
I wonder if it should simply bail out updating. If it wouldn't be sufficient, would you please add more tests to fail?

Reply:
Maybe something like this:
```js
const promise = Promise.resolve()
const countAtom = atom(0)
const derivedAtom = atom((get) => {
  get(countAtom)
  return promise
})
```

Reply:
After reading the comment https://github.com/pmndrs/jotai/pull/1933#discussion_r1193074433,
I don't think we can support the above case, or we shouldn't as dependencies could have been changed.